src/app/api/tools/route.ts

```diff
@@ -1,49 +1,115 @@
-import { queryTools } from '@/lib/utils/firestore';
+import { prismaClient } from '@bitte-ai/data';
 import { NextRequest, NextResponse } from 'next/server';
-import { Tool } from '@/lib/types/tool.types';
-import { kv } from '@vercel/kv';
-import { BittePrimitiveNames } from '@/lib/constants';
 
-const getPingsByTool = async (toolName: string): Promise<number | null> => {
-  return await kv.get<number>(`smart-action:v1.0:tool:${toolName}:pings`);
+const { BITTE_API_URL = 'https://wallet.bitte.ai/api/v1' } = process.env;
+
+type SqlTool = {
+  pings: number;
+  isPrimitive: boolean;
+  image: string;
+  function: object;
+  id: string;
+  agentId: string | null;
+  execution: object | null;
+  type: string;
+  verified: boolean;
+  chainIds: string[];
+};
+
+type PrimitiveResponse = {
+  id: string;
+  function: object;
+  chainIds: string[];
+  pings: number;
 };
 
 export async function GET(request: NextRequest) {
+  // Get search parameters
   const { searchParams } = new URL(request.url);
   const functionName = searchParams.get('function');
   const verifiedOnly = searchParams.get('verifiedOnly') !== 'false';
   const offset = parseInt(searchParams.get('offset') || '0');
+  const limit = parseInt(searchParams.get('limit') || '10000');
   const chainId = searchParams.get('chainId');
 
   try {
-    const tools = await queryTools<Tool>({
-      verified: verifiedOnly,
-      functionName: functionName || undefined,
-      offset,
-      chainId: chainId || undefined,
-    });
-
-    const toolsWithPrimitiveFlags = tools.map((tool) => ({
-      ...tool,
-      isPrimitive: BittePrimitiveNames.includes(tool.function.name),
-      image: BittePrimitiveNames.includes(tool.function.name)
-        ? `/bitte-symbol-black.svg`
-        : tool.image,
-    }));
+    let whereConditions = [];
+    let params = [];
+
+    if (verifiedOnly) {
+      whereConditions.push(`a.verified = true`);
+    }
+
+    if (functionName) {
+      whereConditions.push(`t.function->>'name' = $${functionName}`);
+      params.push(functionName);
+    }
+
+    if (chainId) {
+      whereConditions.push(`$${params.length + 1} IN a.chain_ids`);
+      params.push(chainId);
+    }
```

Comment:
Not sure I understand this formatted string. I would expect to see `chainId IN a.chain_ids`. Something like:

```suggestion
    if (chainId) {
      whereConditions.push(`${chainId} IN a.chain_ids`);
      params.push(chainId);
    }
```

OH.... I see you are appending `chainId` to the params and using the notation `$N`. 

This seems a bit inconsistent with how the above where conditions are treated.

Reply:
It's the same on all the conditions. If they have a param, we use params. We could skip them, possibly opening up avenues for SQL injection.